app/models/wiki.rb

```diff
@@ -5,4 +5,16 @@ class Wiki < ActiveRecord::Base
   validates :body, length: { minimum: 20 }, presence: true
 
   validates :user, presence: true
+
+  # check to see if a user is present of signed in.
+  # If it is a premium or admin user, show all. Else, show all wikis that are public
+  scope :visible_to, -> (user){ user && ((user.premium?) || (user.admin?)) ? where(private: false) + where(private: true, user: user) : where(private: false) }
```

Comment:
I would probably make this a method instead of a scope, but I know this is for learning purposes! There is just a lot of complicated logic in this scope.